add first public versions of mascon programs

tsutterley opened the pull request:
add documentation outlining time series programs
add documentation outlining grace/grace-fo processing

yaramohajerani reviewed (commented):
The documentation is very clear and thorough, but I have the following comments to make the code and the documentation clearer and more user-friendly:

- It would be helpful if the documentation explains more clearly how all the different reference frames should be used. For example, the default reference for `read_love_numbers` is CE, whereas the default for `calc_mascon` is CF. It would be helpful to guide the user through what parameters they should choose for going from standard geoid harmonics to mass balance time-series for mascons.
- `calc_mascon` requries the user to provide both the GIA name and the GIA input file. However, the name is not the same as the GIA title string used in the file name. It makes it a little clunky to load the GIA data every time to get the title for the file. Would help to either make the gia name and the title string in file name equivalent, or make a similar module for just retrieving the title from the name without needing the input file and loading the data from file.
- As a longer-term goal, it would be useful to have a script that takes the kernels produced by `calc_sensitvity_kernel.py` and evaluates the time-series given GRACE harmonic data, instead of re-calculating the kernels every time in `calc_mascon.py`. 
- It would be helpful to briefly discuss why there is different options for the love numbers and how the user should choose (or at least what is new in the later solutions, if possible to explain briefly.)

tsutterley commented:
> * It would be helpful if the documentation explains more clearly how all the different reference frames should be used. For example, the default reference for `read_love_numbers` is CE, whereas the default for `calc_mascon` is CF. It would be helpful to guide the user through what parameters they should choose for going from standard geoid harmonics to mass balance time-series for mascons.

This is a good point.  I added blurbs to talk more about reference frames and why one should be chosen over the other.  I also added blurbs saying that the reference frame needs to be changed for the load Love numbers if geocenter solutions are included.

> * `calc_mascon` requries the user to provide both the GIA name and the GIA input file. However, the name is not the same as the GIA title string used in the file name. It makes it a little clunky to load the GIA data every time to get the title for the file. Would help to either make the gia name and the title string in file name equivalent, or make a similar module for just retrieving the title from the name without needing the input file and loading the data from file.

This is another good point.  Generalizing things for the public release did make this a little clunky with the files needing to be read every time.  I may add an option to `read_GIA_harmonics.py` to stop it from reading the files for programs where the harmonics aren't needed.

> * As a longer-term goal, it would be useful to have a script that takes the kernels produced by `calc_sensitivity_kernel.py` and evaluates the time-series given GRACE harmonic data, instead of re-calculating the kernels every time in `calc_mascon.py`.

I'll leave this as a to do.  You're definitely right things could be optimized if the parameters aren't being changed each time.

> * It would be helpful to briefly discuss why there is different options for the love numbers and how the user should choose (or at least what is new in the later solutions, if possible to explain briefly.)

I added blurbs to talk about load Love numbers and why there are options now.
